src/workloads/query/GroupLikeDistinct.yml

```diff
@@ -0,0 +1,169 @@
+    Threads: 32
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: {unused: "please specify in which phases this actor should be active."}}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Duration: 30 seconds
+          Collection: *Collection
+          Operations:
+          - OperationMetricsName: DistinctExecutionMetrics
+            OperationName: aggregate
+            OperationCommand:
+              Pipeline: {^Parameter: {Name: "Pipeline", Default: []}}
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: 5000
+        BatchSize: 1000
+        Document:
+          _id: {^Inc: {start: 0, step: 1}}
+          a: {^RandomInt: {min: 0, max: 500}}
+          b: {^RandomInt: {min: 0, max: 500}}
+          c: {^RandomInt: {min: 0, max: 500}}
+        Indexes:
+          - keys: {a: 1, b: 1, c: 1}
+
+# The following use a DISTINCT_SCAN.
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroup"
+      OnlyActiveInPhase: 2
+      Pipeline: [{$group: {_id: "$a"}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSort"
+      OnlyActiveInPhase: 3
+      Pipeline: [{$sort: {a: 1}}, {$group: {_id: "$a"}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSortFirst"
+      OnlyActiveInPhase: 4
+      Pipeline: [{$sort: {a: 1, b: 1}}, {$group: {_id: "$a", accum: {$first: "$b"}}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSortFirstFullDocument"
+      OnlyActiveInPhase: 5
+      Pipeline: [{$sort: {a: 1, b: 1, c: 1}}, {$group: {_id: "$a", accum: {$first: "$$ROOT"}}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSortLast"
+      OnlyActiveInPhase: 6
+      Pipeline: [{$sort: {a: -1, b: -1}}, {$group: {_id: "$a", accum: {$last: "$b"}}}]
+
+- ActorFromTemplate:
+    TemplateName: RunAggCommand
+    TemplateParameters:
+      Name: "DistinctGroupSortLastFullDocument"
+      OnlyActiveInPhase: 7
+      Pipeline: [{$sort: {a: 1, b: 1, c: 1}}, {$group: {_id: "$a", accum: {$last: "$$ROOT"}}}]
+
+# The following do not yet use DISTINCT_SCAN, but they will after SERVER-84347.
```

Comment:
I'd rather avoid having to update the workload after SERVER-84347 is merged. Especially, changing the phase names as it would reset the history trends. The comments and naming should reflect the scenarios the workload covers, not the optimizations the query engine might or might not do (e.g. if the workload is run against prior versions of mongo or we have to disable the feature, or whatever). Thus, I'd suggest to either drop "Distinct" from the phase names above or use it the phase names below.